File: app/src/main/java/com/walhalla/appextractor/adapter/viewholder/LogViewHolder.java
package com.walhalla.appextractor.adapter.viewholder;

import android.view.View;

import com.walhalla.appextractor.databinding.ItemLoggerBinding;

import com.walhalla.appextractor.model.LogViewModel;

public class LogViewHolder extends BaseVh<LogViewModel> implements View.OnClickListener {

    private final LoggerAdapter.ChildItemClickListener listener;
    private final ItemLoggerBinding mBinding;

    public LogViewHolder(ItemLoggerBinding binding, LoggerAdapter.ChildItemClickListener listener) {
        super(binding.getRoot());
        this.mBinding = binding;
        this.mBinding.getRoot().setOnClickListener(this);
        this.listener = listener;

        //Log.d("@hash=" + this.hashCode());
    }


    @Override
    public void onClick(View view) {
        listener.onClick0(view, getAdapterPosition());
    }

    @Override
    public void bind(LogViewModel operation, int position) {
        if (operation != null) {
            mBinding.text1.setText(operation.getText());
            mBinding.imageView1.setImageResource(operation.getIcon());
        }
    }
}

File: app/src/main/java/com/walhalla/appextractor/adapter/viewholder/LogViewHolder.kt
package com.walhalla.appextractor.adapter.viewholder

import android.view.View
import com.walhalla.appextractor.adapter.viewholder.LoggerAdapter.ChildItemClickListener
import com.walhalla.appextractor.databinding.ItemLoggerBinding
import com.walhalla.appextractor.model.LogViewModel

class LogViewHolder(private val mBinding: ItemLoggerBinding, listener: ChildItemClickListener) :
    BaseVh<LogViewModel>(mBinding.root), View.OnClickListener {
    private val listener: ChildItemClickListener

    init {
        mBinding.root.setOnClickListener(this)
        this.listener = listener

        //Log.d("@hash=" + this.hashCode());
    }


    override fun onClick(view: View) {
        listener.onClick0(view, adapterPosition)
    }

    override fun bind(operation: LogViewModel, position: Int) {
        if (operation != null) {
            mBinding.text1.text = operation.text
            mBinding.imageView1.setImageResource(operation.icon)
        }
    }
}
